pallets/asset/src/lib.rs

```diff
@@ -1856,6 +1861,39 @@ impl<T: Trait> Module<T> {
         identifiers: Vec<AssetIdentifier>,
         funding_round: Option<FundingRoundName>,
     ) -> DispatchResult {
+        let create = || {
+            Self::base_create_asset(
+                origin,
+                name,
+                ticker,
+                total_supply,
+                divisible,
+                asset_type,
+                identifiers,
+                funding_round,
+            )
+        };
+        // Mint total supply to PIA
+        if total_supply > Zero::zero() {
+            with_transaction(|| {
+                let (sender, did) = create()?;
+                Self::_mint(&ticker, sender, did, total_supply, None)
+            })
+        } else {
+            create().map(drop)
+        }
```

Comment:
Very nit picky and opinionated suggestion but this makes the code more readable IMO. It does cause an additional storage wrapper to be created but that makes no practical difference.

Feel free to apply or drop.

```suggestion
        with_transaction(|| {
            let (sender, did) = Self::base_create_asset(
                origin,
                name,
                ticker,
                total_supply,
                divisible,
                asset_type,
                identifiers,
                funding_round,
            )?;
            if total_supply > Zero::zero() {
                Self::_mint(&ticker, sender, did, total_supply, None)
            }
            Ok(())
        }
```